### DISCUSSION TITLE
Ability to play a custom Totem animation

### DISCUSSION THREAD

**Discussion by Codingale** [PurpurMC/Purpur#1282] [Feature Requests]
`Player#playTotemAnimation(@Nullable int CustomModelData)` (Edit: Pardon the mention if that account is active)

I'm unsure if it's possible to add in a general ItemStack input but I've seen hacks using NMS to use SetCustomModelData and resource packs for work arounds, so that would be very appreciated.

if this already exists please let me know, I'm half asleep writing after powering through random nonsense code I wrote.. 

**edit** after double checking I forgot to change the suggested method name, it should be correct now

**Comment by RhythmicSys** [comment]
I'm a bit confused what this is asking. If you're asking to modify the normal totem animation, that would be done with a resource pack, as far as I know, otherwise this feature request is currently too vague

**Comment by Codingale** [comment]
I forgot I even made this at 3 am.... ironically I'm up again at 3 AM. I was saying how you can use it, optionally with `CustomModelData`, basically you can use NMS to send packets to make a client play the animation anytime, and if I recall you can override the item shown, or at very least the `CustomModelData` that is sent... 

I was half asleep when I wrote this but there was a plugin somewhere that made me think it would be a good feature overall for say picking up items for quests etc.. I think.

Basically the request is to just have the NMS brought to the main API layer
